### DISCUSSION TITLE
Option to expire SSH signing keys

### DISCUSSION THREAD

**Discussion by mrdomino** [community/community#110433] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

It'd be good to have the option to set a previously uploaded SSH signing key as "expired," so that previously signed commits with that key will still show as "Verified" but no new signatures pushed to Github after the time of expiry will be accepted.

Right now it seems that the only two options for a key are "keep it in your profile, so that it can still be actively used" and "remove it entirely, erasing the previously recorded verification information for any commits signed with it." If a key is sunsetted or compromised, neither of these two options are great.

**Comment by rhysperry111** [comment]
Was (un)pleasantly surprised that this isn't a thing today when I went to rotate all my keys. I don't think it's really an option for me to effectively mark all of my verified commits as unverified, but I also feel pretty uncomfortable leaving an old key attached to my account.
